doc/etcd-defrag-cronjob.yaml

```
@@ -0,0 +1,51 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: etcd-defrag
+  namespace: kube-system
+spec:
+  schedule: "14 9 * * 1-5"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+          - name: etcd-defrag
+            image: ghcr.io/ahrtr/etcd-defrag:v0.4.0
+            args:
+            - --cacert=/etc/kubernetes/pki/etcd/ca.crt
+            - --cert=/etc/kubernetes/pki/etcd/peer.crt
+            - --key=/etc/kubernetes/pki/etcd/peer.key
```

Usually etcd-defrag uses the same certificates as etcdctl, so suggest to change the name to make it more explicitly (or change to `client.crt/client.key`). Also let's explicity add an endpoint
```suggestion
            - --endpoints https://127.0.0.1:22379
            - --cacert=/etc/kubernetes/pki/etcd/ca.crt
            - --cert=/etc/kubernetes/pki/etcd/etcdctl.crt
            - --key=/etc/kubernetes/pki/etcd/etcdctl.key
```

---

The certificate generated by kubeadm (e.g. peer.crt, peer.key) are used by etcd to communicate with other etcd peers. But the role of etcd-defrag [against the etcd] is a client side [similar to etcdctl], and the etcd is the server side. The best practice should generate dedicate certificate for this tool, but from pragmatic perspective, we can reuse the certificate prepared for etcdctl.

---

Just as I mentioned, the best practice should generate dedicated certificate for this tool. But at least `apiserver-etcd-client.crt` is better than `peer.crt` in this case.